### PR TITLE
Upgrade CGAL to 5.0.2

### PR DESCRIPTION
**Pull request by maxGimeno**
This PR upgrades CGAL to the version 5.0.2.

### CODE REVIEW

File: ports/cgal/portfile.cmake [hunk 1]
```diff
@@ -5,8 +5,8 @@ vcpkg_buildpath_length_warning(37)
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO CGAL/cgal
-    REF releases/CGAL-5.0.1
-    SHA512 656e207bdc7003dd2e3e4b31b33f9ac3af5074e4892dda645f67fccdc0ac22982d8bf3c62f9c556847ba0447c14f79923712fc6320f8c10ea77e17bf77eb3e2f
+    REF releases/CGAL-5.0.2
+    SHA512 108f1d6f68674e123fd90143049f30a7e7965827468828f75ba7ae0b7ba174690520bafdf0648853c1b28895d6a9a0c7349c03e678c13395a84ffe7397c97e99
     HEAD_REF master
 )
```
Comment (LilyWangL) [inline ports/cgal/portfile.cmake]:
```
set(WITH_CGAL_Qt5  OFF)
if("qt" IN_LIST FEATURES)
  set(WITH_CGAL_Qt5 ON)
endif()
```
The way to handle feature can be replaced by vcpkg_check_features().

File: ports/cgal/portfile.cmake [hunk 2]
```diff
@@ -5,8 +5,8 @@ vcpkg_buildpath_length_warning(37)
 vcpkg_from_github(
```
Comment (LilyWangL) [inline ports/cgal/portfile.cmake]: `include(vcpkg_common_functions)` is deprecated. Please remove it.

**Review by LilyWangL — CHANGES_REQUESTED** [review]

**Comment by LilyWangL** [comment]
@maxGimeno Thanks for your PR. Please fix this changes as below:
1. Can you please use command `file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)` to handle copyright in **portfile.cmake L49**?
2. `vcpkg_test_cmake` in **portfile.cmake L62** is deprecated. Please remove it.

**Comment by maxGimeno** [comment]
@LilyWangL If I didn't miss anything it should be OK now.

**Comment by LilyWangL** [comment]
> @LilyWangL If I didn't miss anything it should be OK now.

Thanks for your changes. Could you please use command `file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)` to handle copyright instead of `file(COPY ${SOURCE_PATH}/Installation/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/cgal) file(RENAME ${CURRENT_PACKAGES_DIR}/share/cgal/LICENSE ${CURRENT_PACKAGES_DIR}/share/cgal/copyright)`?

**Comment by maxGimeno** [comment]
Ah yes, I did forget something. 
Done.

**Review by LilyWangL — APPROVED** [review]

File: ports/cgal/portfile.cmake [hunk 3]
```diff
@@ -46,8 +45,8 @@ endif()
 
 file(WRITE ${CURRENT_PACKAGES_DIR}/lib/cgal/CGALConfig.cmake "include (\$\{CMAKE_CURRENT_LIST_DIR\}/../../share/cgal/CGALConfig.cmake)")
 
-file(COPY ${SOURCE_PATH}/Installation/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/cgal)
-file(RENAME ${CURRENT_PACKAGES_DIR}/share/cgal/LICENSE ${CURRENT_PACKAGES_DIR}/share/cgal/copyright)
+
+file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
```
Comment (LilyWangL) [inline ports/cgal/portfile.cmake]: Sorry, this command should be `file(INSTALL ${SOURCE_PATH}/Installation/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)`.

**Review by LilyWangL — CHANGES_REQUESTED** [review]

File: ports/cgal/portfile.cmake (hunk 3 above)
Comment (lrineau) [inline ports/cgal/portfile.cmake]:
```suggestion
file(INSTALL ${SOURCE_PATH}/Installation/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
```

**Comment by strega-nil** [comment]
thanks @maxGimeno :)

**Pull request merged**